### PR TITLE
GUT-102: create version of CSS for high-specificity sites

### PR DESCRIPTION
**Pull request by ethitter**
Expecting that #306 will land before this, I haven't added the new build files yet since the filenames will change.

### CODE REVIEW

File: packages/larva/scripts/config/gulpfile.js [hunk 1]
```diff
@@ -46,46 +47,82 @@ const stylelint = ( file ) => {
...
-			.pipe( gulp.dest( cssDest ) );
+		postCssPlugins.push( postCssNoop );
 	}
 
+	gulp.src( './entries/*.scss' )
+		.pipe( gulpStylelint( stylelintOpts ) )
+		.pipe( sass( sassOpts ).on( 'error', sass.logError ) )
+		.pipe( gulpPostCss( postCssPlugins ) )
+		.pipe( gulp.dest( cssDest ) )
+		.pipe( gulpPostCss( [ declareImportanceForAll ] ) )
```
Comment (laras126) [inline packages/larva/scripts/config/gulpfile.js:122]: This gulpfile is used in themes to build the local CSS as well, so they will also generate an `XX-important` stylesheet for each entry even when its not needed – do you think we can have a flag or separate gulp script so this only happens when required? There are some utilities that exist to get CLI args that can be added to the prod-scss.js script.

**Comment by ethitter** [comment]
@laras126 This is ready for another look!

**Review by laras126 — CHANGES_REQUESTED** [review]
Yayy this looks so great! Two small changes:
1. Add a note to [the Larva package README](https://github.com/penske-media-corp/pmc-larva/blob/master/packages/larva/README.md#larva-binary--build-scripts) about this feature
2. Update the gulp-sass compiler to use Dart Sass (this is a small scope increase for this feature – LMK if an issue) – see [the gulpfile.js in packages/larva-css](https://github.com/penske-media-corp/pmc-larva/blob/master/packages/larva-css/gulpfile.js#L19) for an example, and [this section of the gulp-sass README](https://www.npmjs.com/package/gulp-sass#basic-usage).

Thanks!!

**Comment by ethitter** [comment]
@laras126 I made those two updates. There's a bit of churn in the `package.json` and `package-lock.json` files due to the version bump.

I also updated the `.gitignore` to exclude the "important variants" that are copied into the larva package from the root `build` directory. The ticket didn't mention including them there (only in the root directory), and I didn't think they were useful in there.

**Comment by laras126** [comment]
> I also updated the `.gitignore` to exclude the "important variants" that are copied into the larva package from the root `build` directory. The ticket didn't mention including them there (only in the root directory), and I didn't think they were useful in there.

Let's see...I think we cannot ignore those from packages/larva. We probably could exclude the copied ones in the root build since those are only relevant for local dev, but until we are able to prioritize figuring out how to install the package from root, we will need them in packages/larva/build so they are available to projects that install them.

**Review by laras126 — CHANGES_REQUESTED** [review]
See other comment – unfortunately, I don't think we can't ignore the copied files since consuming projects will access them from packages/larva/build until we can figure out installing from root.

**Comment by ethitter** [comment]
@laras126 Understood, will fix that momentarily. 

Another thought I had: we don't need an `!important` version of `larva-ui.css`, do we?

**Comment by laras126** [comment]
> Another thought I had: we don't need an `!important` version of `larva-ui.css`, do we?

@ethitter Correct, we do not! I see that it is not in the PR – did you delete manually after the build? 

Just noting that it might be accidentally committed / confusing for devs in the future, though I definitely don't think there should be code to specifically ignore it – so, even if we don't need it, my instinct is that it's a lesser evil to have it in the repo vs. manually removed. But lmk if I'm missing something!

**Review by laras126 — APPROVED** [review]
Wonderful! See comment about the larva-ui stylesheet, otherwise this is A+ fantastic!!

**Comment by ethitter** [comment]
I added an exclusion in the gulpfile that skips the `larva-ui.css` when generating the important variants, so no risk of devs committing it accidentally.

**Pull request merged**